bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/LaunchConfigFinder.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright 2024-2025 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.util;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.ILaunchManager;
+import org.eclipse.launchbar.core.ILaunchDescriptor;
+
+public class LaunchConfigFinder
+{
+	private final ILaunchManager launchManager;
+
+	public LaunchConfigFinder()
+	{
+		launchManager = DebugPlugin.getDefault().getLaunchManager();
+	}
+
+	public LaunchConfigFinder(ILaunchManager launchManager)
+	{
+		this.launchManager = launchManager;
```

Comment:
It would be appropriate to call the DebugPlugin.getDefault().getLaunchManager() method directly within the findAppropriateLaunchConfig() function, as there won't be multiple instances of it.

With that we could remove class constructors

Reply:
done